some test failures on the one that I modified:

```
[camel-milvus] [ERROR] org.apache.camel.component.milvus.it.MilvusComponentIT.search -- Time elapsed: 1.128 s <<< FAILURE!
java.lang.AssertionError: 

Expected size: 2 but was: 1 in:
[[[score:8.580661, userID:467082913052361728, userAge:5],
    [score:10.604269, userID:467082913052361727, userAge:63]]]
	at org.apache.camel.component.milvus.it.MilvusComponentIT.search(MilvusComponentIT.java:216)
[camel-milvus] [ERROR] org.apache.camel.component.milvus.it.MilvusComponentIT.delete -- Time elapsed: 0.140 s <<< FAILURE!
java.lang.AssertionError: 

Expecting empty but was: [[]]
	at org.apache.camel.component.milvus.it.MilvusComponentIT.delete(MilvusComponentIT.java:275)
[camel-milvus] [INFO] Running org.apache.camel.component.milvus.it.MilvusComponentIT
[camel-milvus] [ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.174 s <<< FAILURE! -- in org.apache.camel.component.milvus.it.MilvusComponentIT
[camel-milvus] [ERROR] org.apache.camel.component.milvus.it.MilvusComponentIT.search -- Time elapsed: 0.039 s <<< FAILURE!
java.lang.AssertionError: 

Expected size: 2 but was: 1 in:
[[]]
	at org.apache.camel.component.milvus.it.MilvusComponentIT.search(MilvusComponentIT.java:216)
[camel-milvus] [ERROR] org.apache.camel.component.milvus.it.MilvusComponentIT.delete -- Time elapsed: 0.128 s <<< FAILURE!
java.lang.AssertionError: 

Expecting empty but was: [[]]
	at org.apache.camel.component.milvus.it.MilvusComponentIT.delete(MilvusComponentIT.java:275)
[camel-milvus] [INFO] Running org.apache.camel.component.milvus.it.MilvusComponentIT
[camel-milvus] [ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.191 s <<< FAILURE! -- in org.apache.camel.component.milvus.it.MilvusComponentIT
[camel-milvus] [ERROR] org.apache.camel.component.milvus.it.MilvusComponentIT.search -- Time elapsed: 0.037 s <<< FAILURE!
java.lang.AssertionError: 

Expected size: 2 but was: 1 in:
[[]]
	at org.apache.camel.component.milvus.it.MilvusComponentIT.search(MilvusComponentIT.java:216)
[camel-milvus] [ERROR] org.apache.camel.component.milvus.it.MilvusComponentIT.delete -- Time elapsed: 0.147 s <<< FAILURE!
java.lang.AssertionError: 

Expecting empty but was: [[]]
	at org.apache.camel.component.milvus.it.MilvusComponentIT.delete(MilvusComponentIT.java:275)
```

but they passed locally... investigating

EDIT: I guess I launched the test without it recompiled correctly. I was reproducing. The problem was that it is a List of List and we need to look inside of the first list

--- a/components/camel-ai/camel-milvus/src/test/java/org/apache/camel/component/milvus/it/MilvusComponentIT.java
+++ b/components/camel-ai/camel-milvus/src/test/java/org/apache/camel/component/milvus/it/MilvusComponentIT.java
@@ -213,7 +213,7 @@ public class MilvusComponentIT extends MilvusTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(result.getMessage().getBody(SearchResponse.class).getRowRecords().size() == 2);
+        assertThat(result.getMessage().getBody(SearchResponse.class).getRowRecords().get(0)).hasSize(2);
     }
 
     @Test
@@ -234,7 +234,7 @@ public class MilvusComponentIT extends MilvusTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(result.getMessage().getBody(QueryResults.class).getFieldsDataCount() == 2);
+        assertThat(result.getMessage().getBody(QueryResults.class).getFieldsDataCount()).isEqualTo(2);
     }
 
     @Test
@@ -272,7 +272,7 @@ public class MilvusComponentIT extends MilvusTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(result.getMessage().getBody(SearchResponse.class).getRowRecords().size() == 0);
+        assertThat(result.getMessage().getBody(SearchResponse.class).getRowRecords().get(0)).isEmpty();
     }
 
     // --- Helper-based tests (same operations via MilvusHelper beans) ---

--- a/components/camel-ai/camel-pinecone/src/test/java/org/apache/camel/component/pinecone/it/PineconeComponentIT.java
+++ b/components/camel-ai/camel-pinecone/src/test/java/org/apache/camel/component/pinecone/it/PineconeComponentIT.java
@@ -108,7 +108,7 @@ public class PineconeComponentIT extends CamelTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(!result.getMessage().getBody(QueryResponseWithUnsignedIndices.class).getMatchesList().isEmpty());
+        assertThat(result.getMessage().getBody(QueryResponseWithUnsignedIndices.class).getMatchesList()).isNotEmpty();
     }
 
     @Test
@@ -132,7 +132,7 @@ public class PineconeComponentIT extends CamelTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(!result.getMessage().getBody(QueryResponseWithUnsignedIndices.class).getMatchesList().isEmpty());
+        assertThat(result.getMessage().getBody(QueryResponseWithUnsignedIndices.class).getMatchesList()).isNotEmpty();
     }
 
     @Test
@@ -148,7 +148,7 @@ public class PineconeComponentIT extends CamelTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(!result.getMessage().getBody(QueryResponseWithUnsignedIndices.class).getMatchesList().isEmpty());
+        assertThat(result.getMessage().getBody(QueryResponseWithUnsignedIndices.class).getMatchesList()).isNotEmpty();
     }
 
     @Test
@@ -179,7 +179,7 @@ public class PineconeComponentIT extends CamelTestSupport {
 
         assertThat(result).isNotNull();
         assertThat(result.getException()).isNull();
-        assertThat(result.getMessage().getBody(FetchResponse.class).getVectorsCount() != 0);
+        assertThat(result.getMessage().getBody(FetchResponse.class).getVectorsCount()).isNotZero();
     }
 
 }

--- a/components/camel-diagram/src/test/java/org/apache/camel/diagram/DefaultRouteDiagramDumperTest.java
+++ b/components/camel-diagram/src/test/java/org/apache/camel/diagram/DefaultRouteDiagramDumperTest.java
@@ -76,7 +76,7 @@ class DefaultRouteDiagramDumperTest extends CamelTestSupport {
         RouteDiagramDumper dumper = PluginHelper.getRouteDiagramDumper(context);
         dumper.dumpRoutesToFile("*", RouteDiagramDumper.Theme.DARK, f);
 
-        assertThat(f.exists());
+        assertThat(f).exists();
     }
 
     @Test
@@ -89,7 +89,7 @@ class DefaultRouteDiagramDumperTest extends CamelTestSupport {
         RouteDiagramDumper dumper = PluginHelper.getRouteDiagramDumper(context);
         dumper.dumpRoutesToFolder("*", RouteDiagramDumper.Theme.LIGHT, f);
 
-        assertThat(f.exists());
+        assertThat(f).exists();
         assertThat(f.list()).contains("myRoute.png", "otherRoute.png");
     }
 

